providers/song-info-front.js

```diff
@@ -63,3 +65,23 @@ function setupTimeChangeListener() {
 	});
 	progressObserver.observe($('#progress-bar'), { attributeFilter: ["value"] })
 }
+
+function setupRepeatChangeListener() {
+	const repeatObserver = new MutationObserver(mutations => {
+		ipcRenderer.send('repeatChanged', mutations[0].target.title);
+	});
+	repeatObserver.observe($('#right-controls .repeat'), { attributeFilter: ["title"] });
+
+	// Emit the initial value as well; as it's persistent between launches.
+	ipcRenderer.send('repeatChanged', $('#right-controls .repeat').title);
+}
+
+function setupVolumeChangeListener() {
+	const volumeObserver = new MutationObserver(mutations => {
+		ipcRenderer.send('volumeChanged', mutations[0].target.value);
+	});
+	volumeObserver.observe($('#right-controls .volume-slider'), { attributeFilter: ["value"] });
```

Comment:
you should observe `$('video').volume` which is much more accurate

(I think that changing increments of 0.01 via precise-volume will not trigger volume slider changes which will make the mpris player not have the precise value)

Reply:
That doesn't appear to give the correct values?

Reply:
I'm pretty sure that `$('#right-controls .volume-slider').value` doesn't give the right volume and `$('video').volume` is the correct one

you can easily test this by changing `$('video').volume` and you'll see the video volume changes, while manually changing the volume slider value wont change the video volume

maybe you got confused because its a different format? 0.01/1 instead of 1/100

Reply:
> maybe you got confused because its a different format? 0.01/1 instead of 1/100

Yea, in my quick test I got 0.01 instead of 100. I'll look into it again.

Reply:
haha yeah so you can just `*100` it

Reply:
I can use `$('video').volume` to get the initial volume, but can't get events to trigger when it fires. I've tried the `MutationObserver` with a filter for `volume`, and the `volumechanged` event handler on the video element.

Is there anything that changes the volume without affecting the slider? Because there shouldn't be, right?

Reply:
> Is there anything that changes the volume without affecting the slider? Because there shouldn't be, right?

No and there shouldn't be, but its possible so best be avoided if possible

> volumechanged event handler on the video element

that should work, I just tested on my side and it works, can you show me your code?

if you write this in devtools:
```js
$('video').addEventListener('volumechange', console.log);
```
doesn't it write a message on every change?

Reply:
Okay, I don't know why it didn't work; it does now.

```js
$('video').addEventListener('volumechange', (_) => {
    ipcRenderer.send('volumeChanged', $('video').volume * 100);
});
ipcRenderer.send('volumeChanged', $('video').volume * 100);
```

However, instead of 100 or 30, I'll get values like 98.855310546902 or 29.656593164070603

Reply:
Then it's still 99 instead of 100? Why is this more inaccurate?

Reply:
Okay, I've implemented it that way; works.